test/basic.html

```diff
@@ -55,34 +55,48 @@
           assert.equal(collapse.horizontal, false);
         });
 
-        test('default opened height', function(done) {
-          setTimeout(function() {
-            // store height
-            collapseHeight = getComputedStyle(collapse).height;
-            // verify height not 0px
-            assert.notEqual(collapseHeight, '0px');
-            done();
-          }, delay);
+        test('default opened height', function() {
+          assert.equal(collapse.style.height, 'auto');
         });
 
-        test('set opened to false', function(done) {
+        test('set opened to false triggers animation', function(done) {
           collapse.opened = false;
-          setTimeout(function() {
-            var h = getComputedStyle(collapse).height;
-            // verify height is 0px
-            assert.equal(h, '0px');
+          // animation got enabled
+          assert.notEqual(collapse.style.transitionDuration, '0s');
+          collapse.addEventListener('transitionend', onTransitionEnd);
+
+          function onTransitionEnd() {
+            collapse.removeEventListener('transitionend', onTransitionEnd);
```

Comment:
i don't thiiiiink you need to remove listeners for everything (unless you thing this will be called twice). we certainly don't do this for other tests